eclipse-collections-api/src/main/java/org/eclipse/collections/api/collection/ImmutableCollection.java

```diff
@@ -190,4 +196,37 @@
             Function<? super T, ? extends K> groupBy,
             Function0<? extends V> zeroValueFactory,
             Function2<? super V, ? super T, ? extends V> nonMutatingAggregator);
+
+    /**
+     * @since 9.0
+     */
+    default Stream<T> stream()
+    {
+        return StreamSupport.stream(this.spliterator(), false);
+    }
+
+    /**
+     * @since 9.0
+     */
+    default Stream<T> parallelStream()
+    {
+        return StreamSupport.stream(this.spliterator(), true);
+    }
+
+    /**
+     * @since 9.0
+     */
+    @Override
+    default Spliterator<T> spliterator()
+    {
+        return Spliterators.spliterator(this.iterator(), (long) this.size(), 0);
+    }
+
+    /**
+     * @since 9.0
+     */
+    default Collection<T> castToCollection()
+    {
+        return (Collection<T>) this;
```

Comment:
Yes.  It would be wrong for this to happen though.  The only other options would be for an implementor to return null here, or return some other collection.  This method is being added for symmetry with castToList, castToSet, castToMap.  It is immediately overridden in AbstractImmutableCollection to return this.